src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -157,224 +172,137 @@ public void updatePID() {
-    // feet
-  }
-
-  /// Aimodes
-  public void automode() {
-    adjustHood(
-        ProjectilePrediction(
-            Constants.Shooter.shooterHeight,
-            0,
-            Constants.Shooter.highHeight,
-            getDistance(),
-            Constants.Shooter.gravity,
-            Constants.Shooter.airboneTime)[1]);
-    windFlywheel(
-        (int)
-            (Math.ceil(
-                ProjectilePrediction(
-                    Constants.Shooter.shooterHeight,
-                    0,
-                    Constants.Shooter.highHeight,
-                    getDistance(),
-                    32,
-                    Constants.Shooter.airboneTime)[0])));
-    currentRPM =
-        (int)
-            (Math.ceil(
-                ProjectilePrediction(
-                    Constants.Shooter.shooterHeight,
-                    0,
-                    Constants.Shooter.highHeight,
-                    getDistance(),
-                    32,
-                    Constants.Shooter.airboneTime)[0]));
-  }
-
-  public void lookuptablemode(double distance) {
-    double[] returnarray = lookup(distance);
-    windFlywheel(returnarray[0]);
-    // adjusthood(returnarray[1]);
-  }
-
-  public double[] ProjectilePrediction(
-      double y0, double x0, double y, double x, double g, double t) {
-    x = x + 1; // Applies an offset to target goal center
-    double hoodAngle =
-        Math.toDegrees(
-            Math.atan((y - y0 + 1 / 2 * g * (Math.pow(t, 2))) / x)); // Finds angle to shoot at
-    double velocity =
-        Math.abs(
-            x
-                / (Math.cos(Math.toRadians(hoodAngle))
-                    * t)); // Finds velocity in feet per second to shoot at
-    double[] returnArray = new double[2];
-    returnArray[0] =
-        UnitConversion(velocity, Constants.Shooter.gearDiameter); // Converts FPS to RPM
-    returnArray[1] = hoodAngle;
-    return returnArray;
-  }
-
-  public double UnitConversion(double KBallSpeed, double GearDiameter) {
-    // Convert from FPS of the ball into RPM
-    return (((KBallSpeed * 12) / Constants.Shooter.gearDiameter)
-            * Constants.Shooter.ballFlywheelratio)
-        * 2;
+    return (Constants.Shooter.highHeight - Constants.Shooter.LLHeight) / Math.tan(Math.toRadians((getTY() + Constants.Shooter.LLAngle))); // Return distance in ft
   }
 
   public void prime() {
     // Check what aimMode is active, gets distance if AUTO, winds flywheel, adjusts
     // hood correspondingly
-
-    switch ((int) aimMode) {
-      case 0: // Case for LOW mode, winds flywheel to preset RPM and adjusts hood to preset
-        // angle
-        adjustHood(Constants.Shooter.LOWAngle);
-        windFlywheel(Constants.Shooter.LOWRPM);
-        currentRPM = Constants.Shooter.LOWRPM;
+    switch (aimMode) {
+      case EJECT: // aimMode used to eject unwanted balls from the shooter
+        adjustHood(aimMode.getAngle());
+        windFlywheel(aimMode.getRPM());
```

Comment:
```suggestion
      case EJECT: // aimMode used to eject unwanted balls from the shooter
      case LOW: // aimMode used to dump into the low goal from ~1ft
      case TARMAC: // aimMode used to shoot into the high goal from ~2ft
      case LAUNCH: // aimMode used to shoot into the high goal from the launchpad
        adjustHood(aimMode.getAngle());
        windFlywheel(aimMode.getRPM());
        break;
```

If multiple cases run the same code, you can just list them like this (notice there's not 'break' between them)

Reply:
That's true, I thought switch statements would go on to keep evaluating the others if there was no break, not running the code at the end.